tony-history-server/app/utils/ParserUtils.java

```diff
@@ -0,0 +1,88 @@
+package utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.gson.JsonSyntaxException;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import models.JobConfig;
+import models.JobMetadata;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import play.Logger;
+import play.libs.Json;
+
+import static utils.HdfsUtils.*;
+
+/**
+ * The class handles parsing different file format
+ */
+public class ParserUtils {
+  private static final Logger.ALogger LOG = Logger.of(ParserUtils.class);
+
+  public static JobMetadata parseMetadata(FileSystem fs, Path metricsFilePath) {
```

Comment:
Please add Javadoc to this and other methods. The methods in ParserUtils make assumptions such as the file being JSON or XML that should be documented.